packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -476,27 +583,24 @@ export const optionalFieldEditor: OptionalFieldEditor = {
 			// Should be interpreted as a set of an empty field if undefined.
 			detach: ChangesetLocalId;
 		},
-	): OptionalChangeset => {
-		const moves: Move[] = [[{ localId: ids.fill }, "self", "nodeTargeting"]];
-		const result: OptionalChangeset = {
-			moves,
-			childChanges: [],
-		};
-		if (wasEmpty) {
-			result.reservedDetachId = { localId: ids.detach };
-		} else {
-			moves.push(["self", { localId: ids.detach }, "cellTargeting"]);
-		}
-		return result;
-	},
-
-	clear: (wasEmpty: boolean, detachId: ChangesetLocalId): OptionalChangeset =>
-		wasEmpty
-			? { moves: [], childChanges: [], reservedDetachId: { localId: detachId } }
-			: {
-					moves: [["self", { localId: detachId }, "cellTargeting"]],
-					childChanges: [],
-			  },
+	): OptionalChangeset => ({
+		moves: [],
+		childChanges: [],
+		valueReplace: {
+			isEmpty: wasEmpty,
```

Comment:
this makes me realize that IMO `wasEmpty` is the better name, not a big deal though.

probably worth calling out that emptiness applies to the input context in the doc comment on OptionalChangeType.